pkg/subnet/local.go

```diff
@@ -146,19 +153,65 @@ func (d *SubnetDeployer) doDeploy(chain string, chain_genesis string) error {
 	ctx := binutils.GetAsyncContext()
 
 	ux.Logger.PrintToUser("VM ready. Trying to boot network...")
-	info, err := cli.Start(
+	loadSnapshotOpts := []client.OpOption{
+		client.WithPluginDir(pluginDir),
+		client.WithExecPath(avalancheGoBinPath),
+	}
+	loadSnapshotsInfo, err := cli.LoadSnapshot(
+		ctx,
+		constants.DefaultSnapshotName,
+		loadSnapshotOpts...,
+	)
+	if err != nil {
+		if !strings.Contains(err.Error(), "already bootstrapped") {
+			return fmt.Errorf("failed to start network :%s", err)
+		}
+		ux.Logger.PrintToUser("Network has already been booted. Wait until healthy...")
+	} else {
+		ux.Logger.PrintToUser("Booting Network. Wait until healthy...")
+	}
+
+	d.log.Debug(loadSnapshotsInfo.String())
+
+	subnetIDs, blockchainsInfo, _, err := d.WaitForHealthy(ctx, cli, d.healthCheckInterval)
+	if err != nil {
+		_ = binutils.KillgRPCServerProcess()
+		return fmt.Errorf("failed to query network health: %s", err)
+	}
+
+	// choose next subnetID
+	sort.Strings(subnetIDs)
+	subnetId := ""
+	if len(subnetIDs) > 0 {
+		subnetId = subnetIDs[len(blockchainsInfo)%len(subnetIDs)]
+	}
+
+	blockchainSpecs := []*rpcpb.BlockchainSpec{
+		{
+			VmName:   chain,
+			Genesis:  chain_genesis,
+			SubnetId: &subnetId,
+		},
+	}
+	deployBlockchainsOpts := []client.OpOption{
+		client.WithPluginDir(pluginDir),
+		client.WithBlockchainSpecs(blockchainSpecs),
+	}
+
+	deployBlockchainsInfo, err := cli.DeployBlockchains(
 		ctx,
-		avalancheGoBinPath,
-		opts...,
+		deployBlockchainsOpts...,
 	)
 	if err != nil {
-		return fmt.Errorf("failed to start network :%s", err)
+		return fmt.Errorf("failed to deploy blockchain :%s", err)
 	}
 
-	d.log.Debug(info.String())
-	ux.Logger.PrintToUser("Network has been booted. Wait until healthy. Please be patient, this will take some time...")
+	d.log.Debug(deployBlockchainsInfo.String())
+
+	fmt.Println()
+	ux.Logger.PrintToUser("Blockchain has been deployed. Wail until network acknowledges...")
 
-	endpoints, err := d.WaitForHealthy(ctx, cli, d.healthCheckInterval)
+	_, _, endpoints, err := d.WaitForHealthy(ctx, cli, d.healthCheckInterval)
```

Comment:
This also feels odd, sometimes we check and sometimes we just ignore the returns. It feels like the pattern could be refactored...

Reply:
The code has to wait twice, and use different info for different wait results. What do you sugest for this case?

Reply:
Changed the return of WaitForHealthy to be more general and the callers to create the remaining info